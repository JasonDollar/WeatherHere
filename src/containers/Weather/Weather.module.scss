.container {
  max-width: 1040px;
  margin: 0 auto;
  background-color: var(--color-back);
}

.headerBar {
  width: 100%;
  background-color: #dd0055;
}