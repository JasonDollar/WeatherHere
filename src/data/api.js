const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

