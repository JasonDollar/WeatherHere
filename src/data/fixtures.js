import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}