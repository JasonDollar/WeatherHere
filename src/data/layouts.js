const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



