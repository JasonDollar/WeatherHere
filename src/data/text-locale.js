const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 