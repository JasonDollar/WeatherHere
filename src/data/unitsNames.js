export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}