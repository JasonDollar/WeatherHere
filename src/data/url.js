const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }