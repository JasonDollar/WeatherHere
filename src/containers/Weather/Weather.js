import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import {DARK_URL, MAP_URL} from '../../data/url'
import {DARK_API, MAP_API} from '../../data/api/api'
import axios from 'axios'
import classes from './Weather.module.scss'

import Current from '../../components/Current/Current'
import Daily from '../../components/Daily/Daily'
import Spinner from '../../components/Spinner/Spinner'
import Hourly from '../../components/Hourly/Hourly'
import Updated from '../../components/Updated/Updated'
import ScreenMessage from '../../components/ScreenMessage/ScreenMessage'


class Weather extends Component {
  
  state = {
    forecast: '',
    locationShortName: '',
    error: '',
    isLoading: true,
    updateTime: null,
    address: '',
  }

  componentDidMount() {
    this.parseDataFromLocalStorage()
  }

  componentDidUpdate(prevProps) {
    if ((prevProps.language !== this.props.language)
      || (prevProps.location !== this.props.location) 
      || (prevProps.units !== this.props.units)) 
    {
      this.setState({error: '', isLoading: true})
      this.getWeather(this.props.location.lat, this.props.location.long, this.props.language, this.props.units)
    } 
    if (prevProps.searchValue !== this.props.searchValue && this.props.searchValue !== '') {
      this.setState({error: '', isLoading: true}) 
      this.onNameLocationSearch(this.props.searchValue)
    }
 
  }

  parseDataFromLocalStorage = () => {
    const locationShortName = localStorage.getItem('locationName')
    const location = JSON.parse(localStorage.getItem('locationCoords'))
    if (location) {
      //required to initialize app and search saved location
      this.props.setLocationCoordsToState(location)
      // this.getWeather(location.lat, location.long, this.props.language, this.props.units)
    }
    else if (locationShortName) {
      this.setState({
        locationShortName,
      })
      this.onNameLocationSearch(locationShortName)
    } else {
      this.setState({
        locationShortName: ''
      })
    }
  }

  searchLocationName = (data) => {
    this.setState({
      locationShortName: '',
    })
    let shortName = ''
    let longName = ''

    const placeNames = data.results
    
    console.log(placeNames[0])

    // for(let i = 0; i < placeNames.length; i++) {
    //   for(let j = 0; j < placeNames[i].address_components.length; j++) {
    //     if (placeNames[i].address_components[j].types.includes("locality")) {
    //       shortName = placeNames[i].address_components[j].short_name
    //       longName = placeNames[i].address_components[j].long_name
    //       break
    //     }
    //   }
    // }
    for(let i = 0; i < placeNames[0].address_components.length; i++) {
        if (placeNames[0].address_components[i].types.includes("locality")) {
          shortName = placeNames[0].address_components[i].short_name
          longName = placeNames[0].address_components[i].long_name
          break
      }
    }
    if (shortName) {
      localStorage.setItem('locationName', shortName)
      this.setState({
        address: placeNames[0].formatted_address,
        locationShortName: shortName.length < 5 ? longName : shortName,
      }) 
    }
  }

  getWeather = (lat, long, lang, units) => {
    // returns all places names for given location 
    axios.post(`${MAP_URL}?latlng=${lat},${long}&key=${MAP_API}&language=${lang}`)
      .then(res => res.data)
      .then(data => {
        this.searchLocationName(data)
      })
      .catch(error => this.setState({error: error.message}))

    axios.get(`https://cors-anywhere.herokuapp.com/${DARK_URL}${DARK_API}/${lat},${long}?lang=${lang}&units=${units}`, {
      method: 'HEAD',
      mode: 'no-cors',
    })
      .then(res => {
        this.setState({
        forecast: res.data,
        isLoading: false,
        error: '',
        updateTime: Date.now(),
      })
      
    })
      .catch(error => this.setState({error: error.message}))
  }

  onNameLocationSearch = (location) => {
    axios.post(`${MAP_URL}?address=${location}&key=${MAP_API}`)
      .then(res => {
        return res.data.results[0].geometry.location
      })
      .then(data => {
        const location = {
          lat: data.lat,
          long: data.lng
        }
        this.props.setLocationCoordsToState(location)
        localStorage.setItem('locationCoords', JSON.stringify(location))
      })
      .catch(error => this.setState({error: error.message}))
  }



  render() {
    let component 
    if (this.state.error) {
      component = <ScreenMessage>{this.state.error}</ScreenMessage>
    } else if (!this.state.locationShortName && !this.state.location) {
      component =  <ScreenMessage>{this.props.text.welcome}</ScreenMessage>
    } else if (this.state.geoForbidden && !this.state.location) {
      component =  <ScreenMessage>Not working</ScreenMessage>
    } else if (!this.state.forecast || this.state.isLoading || !this.props.language) {
      component = <Spinner />
    } else {
      component = (
        <Fragment>
          <Current 
            currently={this.state.forecast.currently}
            currentText={this.props.text.current}
            daily={this.state.forecast.daily}
            units={this.state.forecast.flags.units}
            dateText={this.props.text.date}
            timezone={this.state.forecast.timezone}
            locationShortName={this.state.locationShortName}
            
          />
          <Daily 
            daily={this.state.forecast.daily}
            dateText={this.props.text.date}
            timezone={this.state.forecast.timezone}
            units={this.state.forecast.flags.units}
          />
          <Hourly 
            hourly={this.state.forecast.hourly}
            hourlyText={this.props.text.hourly}
            timezone={this.state.forecast.timezone}
            units={this.state.forecast.flags.units}
          />
          <Updated 
            updateTime={this.state.updateTime}
            updateText={this.props.text.update}
            address={this.state.address}
          />
        </Fragment>
      )
    }
    return (
      <main className={classes.container}>
        { component }
      </main>
    );
  }
}

export default Weather;

Weather.propTypes = {
  text: PropTypes.objectOf(PropTypes.shape({
    current: PropTypes.objectOf(PropTypes.string),
    date: PropTypes.objectOf(PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.arrayOf(PropTypes.string),
    ])),
    hourly: PropTypes.string,
    id: PropTypes.string,
    layout: PropTypes.string,
    name: PropTypes.string,
    settings: PropTypes.objectOf(PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.arrayOf(PropTypes.string),
    ])),
    update: PropTypes.string,
  })).isRequired,
  language: PropTypes.string.isRequired,
  searchValue: PropTypes.string.isRequired,
  location: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.number),
  ]),
  setLocationCoordsToState: PropTypes.func.isRequired,
  units: PropTypes.string.isRequired,

}