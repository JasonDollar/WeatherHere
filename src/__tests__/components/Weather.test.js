import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { appTestData as data } from '../../data/fixtures'
import Weather from '../../containers/Weather/Weather'

const testProps = {
  text: data.text.pl,
  language: data.language,
  searchValue: '',
  location: data.coordinates,
  setLocationCoordsToState: jest.fn(),
  units: data.units.si.id,
}

let wrapper

describe('Weather container', () => {
  test('renders component', () => {
    wrapper = <Weather {...testProps} />
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

})