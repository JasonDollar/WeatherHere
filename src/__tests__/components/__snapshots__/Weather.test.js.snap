// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather container renders component 1`] = `
<Weather
  language="pl"
  location={
    Object {
      "lat": 52.2296756,
      "long": 21.0122287,
    }
  }
  searchValue=""
  setLocationCoordsToState={[MockFunction]}
  text={
    Object {
      "current": Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      },
      "date": Object {
        "temperature": "Temperatura",
        "temperatureMax": "Temperatura Max.",
        "temperatureMin": "Temperatura Min.",
        "today": "Dziś",
        "weekDay": Array [
          "Niedziela",
          "Poniedziałek",
          "Wtorek",
          "Środa",
          "Czwartek",
          "Piątek",
          "Sobota",
          "Dziś",
        ],
        "weekDayShort": Array [
          "Nie",
          "Pon",
          "Wt",
          "Śr",
          "Czw",
          "Pt",
          "Sob",
        ],
      },
      "hourly": Object {
        "hour": "Godzina",
        "precipProb": "Możliwość Opadów (%)",
        "precipProbShort": "Możliwość Opadów",
        "temperature": "Temperatura",
        "temperatureShort": "Temperatura",
      },
      "id": "pl",
      "layout": Object {
        "forecast": "Pogoda",
        "geo": "Geo",
        "geoFull": "Geolokalizacja",
        "icon": "Ikona",
        "inputPlaceholder": "Szukaj lokalizacji",
        "search": "Szukaj",
        "settings": "Ustawienia",
      },
      "name": "Polski",
      "settings": Object {
        "changeLang": "Zmień język",
        "changeTheme": "Zmień motyw",
        "changeUnits": "Zmień jednostki",
        "iconName": "Znak wyboru (fajka)",
        "units": Array [
          "SI",
          "US",
          "UK",
          "CA",
        ],
      },
      "update": "Ostatnia aktualizacja: ",
      "welcome": "Szukaj lokalizacji",
    }
  }
  units="si"
/>
`;
